### PR TITLE
Add support for DeleteFactor and GetFactor methods

### PR DESCRIPTION
**Pull request by AshGodfrey**
Add support for delete factor and get factor methods
Add tests for delete factor and get factor methods

### CODE REVIEW

File: pkg/mfa/client.go [hunk 1]
```diff
@@ -153,6 +153,16 @@ type RawVerifyResponse struct {
 	VerifyResponseError
 }
 
+type DeleteFactorOpts struct {
+	// ID of factor to be deleted
+	ID string
```
Comment (blairlunceford) [inline pkg/mfa/client.go:158]: It looks like the convention is to have a more descriptive name - could we use `AuthenticationFactorID` instead for both of these new methods?

File: pkg/mfa/client.go [hunk 2]
```diff
@@ -294,3 +304,80 @@ func (c *Client) VerifyChallenge(
 	}
 
 }
+
+// DeleteFactor deletes a Factor.
```
Comment (blairlunceford) [inline pkg/mfa/client.go:308]: nit: change comment to `Deletes an authentication factor` to be in line with other methods.

File: pkg/mfa/client.go [hunk 3]
```diff
@@ -294,3 +304,80 @@ func (c *Client) VerifyChallenge(
...
+	res, err := c.HTTPClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+
+	return workos_errors.TryGetHTTPError(res)
+}
+
+// GetFactor gets a Factor.
```
Comment (blairlunceford) [inline pkg/mfa/client.go:343]: nit: change comment to `Retrieves an authentication factor` to be in line with other methods.

**Review by blairlunceford — CHANGES_REQUESTED** [review]

**Comment by AshGodfrey** [comment]
Going to need to check why this failing.

**Comment by AshGodfrey** [comment]
Also I switched the ID name as requested, but a lot of our other go functions do just use ID. Not sure if we want to keep it inline with other functions or go with the more descriptive name.

**Comment by blairlunceford** [comment]
> Also I switched the ID name as requested, but a lot of our other go functions do just use ID. Not sure if we want to keep it inline with other functions or go with the more descriptive name.

Thanks! Yes, but this pattern will follow the `ChallengeOpts` and `VerifyOpts` structs in this file, so I think it's good to just make the change here.

**Review by blairlunceford — APPROVED** [review]

**Pull request merged**